jena-arq/src/main/java/org/apache/jena/sparql/core/thrift/IteratorCached.java

```diff
@@ -0,0 +1,21 @@
+package org.apache.jena.sparql.core.thrift;
+
+import java.util.Iterator;
+
+public abstract class IteratorCached<E> implements Iterator<E> {
```

Comment:
Since this class isn't actually caching the contents of the wrapped `Iterator`, perhaps a Javadoc comment explaining its semantics? Presumably subclasses are supposed to do caching of some kind (as below)?